src/main/java/edu/harvard/iq/dataverse/dataaccess/S3AccessIO.java

```diff
@@ -0,0 +1,568 @@
+        ObjectMetadata metadata = new ObjectMetadata();
+        metadata.setContentLength(length);
+        try {
+            s3.putObject(bucketName, key, inputStream, metadata);
+        } catch (SdkClientException ioex) {
+            String failureMsg = ioex.getMessage();
+            
+            if (failureMsg == null) {
+                failureMsg = "S3AccessIO: Unknown exception occured while saving a local InputStream as S3Object";
+            }
+            throw new IOException(failureMsg);
+        }  
+    }
+    
+    //TODO: How do we test this?
+    @Override
+    public List<String>listAuxObjects() throws IOException {
+        String prefix = null;
+        if (!this.canWrite()) {
+            open();
+        }
+        if (dvObject instanceof DataFile) {
+            prefix = s3FileName + ".";
+        } else if (dvObject instanceof Dataset) {
+            prefix = s3FolderPath + "/";
+        }
+                
+        List<String> ret = new ArrayList<>();
+        ObjectListing storedAuxFilesList = null;
+        List<S3ObjectSummary> storedAuxFilesSummary = null;
+        logger.info("S3 prefix: " + prefix);
+
+        ListObjectsRequest req = new ListObjectsRequest().withBucketName(bucketName).withPrefix(prefix);
+        storedAuxFilesList = s3.listObjects(req);
+        storedAuxFilesSummary = storedAuxFilesList.getObjectSummaries();
+        try {
+            while (storedAuxFilesList.isTruncated()) {
+                logger.info("S3 listAuxObjects: going to second page of list");
+                storedAuxFilesList = s3.listNextBatchOfObjects(storedAuxFilesList);
+                storedAuxFilesSummary.addAll(storedAuxFilesList.getObjectSummaries());
+            }
+        } catch (AmazonClientException ase) {
+            logger.warning("Caught an AmazonServiceException in S3AccessIO.listAuxObjects():    " + ase.getMessage());
+            logger.warning("Caught an AmazonServiceException:    " + ase.getMessage());
+        }
+        
+        for (S3ObjectSummary item : storedAuxFilesSummary) {
+            String key = item.getKey();
+            String fileName = key.substring(key.lastIndexOf("/"));
+            logger.info("S3 cached aux object fileName: " + fileName);
+            ret.add(fileName);
+        }
+        return ret;
+    }
+    
+    @Override
+    public void deleteAuxObject(String auxItemTag) {
+        String key = null;
+        if (dvObject instanceof DataFile) {
+            key = s3FileName + "." + auxItemTag;
+        } else if (dvObject instanceof Dataset) {
+            key = s3FolderPath + "/" + auxItemTag;
+        }
+        try {
+            DeleteObjectRequest dor = new DeleteObjectRequest(bucketName, key);
+            s3.deleteObject(dor);
+        } catch (AmazonClientException ase) {
+            logger.warning("S3AccessIO: Unable to delete object    " + ase.getMessage());
+        }
+        
+    }
+    
+    //TODO: is this efficient? i.e. number of calls to S3
```

Comment:
There is the a deleteObjects command that could collapse the number of delete calls to 1.